src/librustc_session/config.rs

```diff
@@ -1705,6 +1705,29 @@ pub fn build_session_options(matches: &getopts::Matches) -> Options {
         );
     }
 
+    if debugging_opts.instrument_coverage {
+        if cg.profile_generate.enabled() || cg.profile_use.is_some() {
+            early_error(
+                error_format,
+                "option `-Z instrument-coverage` is not compatible with either `-C profile-use` \
+                or `-C profile-generate`",
+            );
+        }
+
+        // `-Z instrument-coverage implies:
+        //   * `-Z symbol-mangling-version=v0` - to ensure consistent and reversable name mangling
```

Comment:
It would still be usable without this, just not as nice, right?

Reply:
Since the LLVM coverage tools support analysis across multiple runs (even if the code changes somewhat), the mangled names need to be consistent. I added a comment for clarity.

If your point is, the instrumentation code will work with the old mangler, yes, but IMO it won't work as expected, so I think we should require v0.